signalbot/bot.py

```diff
@@ -306,6 +309,35 @@ def _is_valid_uuid(self, receiver_uuid: str):
         except ValueError:
             return False
 
+    def _is_username(self, receiver_username: str):
+        """
+        Check if username has correct format.
+        Usernames can be between 3 and 32 characters.
+        Usernames may only contain a-z, 0-9 and _.
+        You must include at least two digits at the end of your username.
+        There cannot be more than 9 digits at the of your username.
+        Digits cannnot be 00.
```

Comment:
I like how explicit this is but I think it'd be better to just refer to the source
```suggestion
        Check if username has correct format, as described in https://support.signal.org/hc/en-us/articles/6712070553754-Phone-Number-Privacy-and-Usernames#username_req
        Additionally, cannot have more than 9 digits and the digits cannot be 00.
```